app/persistence/sql/repository.py

```diff
@@ -54,6 +58,59 @@ def __init__(
         self._domain_cls = domain_cls
         self.system = "SQL"
 
+    def _get_relationship_load(
+        self,
+        relationship: QueryableAttribute,
+        preload: list[str] | None = None,
+        depth: int = 1,
+    ) -> _AbstractLoad:
+        """
+        Get the appropriate relationship loading strategy with support for nesting.
+
+        Args:
+            relationship: The relationship attribute to load
+            preload: List of additional relationships to preload
+            depth: Internal tracker for max relationship depth
+
+        Returns:
+            An ORM loading option configured for the relationship
+
+        """
+        load_type = relationship.info.get("load_type", RelationshipLoadType.JOINED)
+        max_recursion_depth = relationship.info.get("max_recursion_depth")
+
+        # Determine the base loading strategy
+        if load_type == RelationshipLoadType.SELECTIN:
+            # Recurse once, we add more loads dynamically below
+            loader = selectinload(relationship, recursion_depth=1)
+        else:
+            loader = joinedload(relationship)
+
+        # This magic ensures we both:
+        # - propagate preloads to self-referential relationships
+        # - recursively join self-referential relationships a configured number of times
+        # Use-case for initial implementation is propagating enhancements etc to
+        # duplicates when preloaded.
+        avoid_propagate: set[str] = set()
+        if back_populates := relationship.info.get("back_populates"):
+            # Don't "bounce back" and form a joining cycle
+            avoid_propagate.add(back_populates)
+        if depth == (max_recursion_depth or 1):
+            # Recursion exit case, maximum length of this relationship's
+            # self-referential chain
+            avoid_propagate.add(relationship.key)
+
+        is_self_referential = relationship.prop.mapper.class_ == self._persistence_cls
+        if preload and is_self_referential:
+            preload = [p for p in preload if p not in avoid_propagate]
+            for nested_rel_name in preload:
+                nested_rel = getattr(relationship.prop.mapper.class_, nested_rel_name)
```

Comment:
This is effectively the same as `getattr(self._persistence_cls, nested_rel_name)`, right?